### DISCUSSION TITLE
Allocating PDCCH and PDSCH in the same slot

### DISCUSSION THREAD

**Discussion by improperaffo** [srsran/srsRAN_Project#465] [General Help] [answered]
Going through `scheduler_time_rr.cpp` we have found these two code sections that hints at the fact that one cannot schedule PDCCH and PDSCH in the same slot.
```
    // UE is already allocated in the PDCCH for this slot (e.g. we should skip a newTx if a reTx has already been
    // allocated for this UE).
    if (res_grid.has_ue_dl_pdcch(ue_cc.cell_index, u.crnti)) {
      return alloc_outcome::skip_ue;
    }
```
and
```
    // UE is already allocated resources.
    if (res_grid.has_ue_ul_pdcch(ue_cc.cell_index, u.crnti)) {
      return alloc_outcome::skip_ue;
    }
```
We have tried going through the specs, but we could not find any reference for this behaviour. May we ask what is it related to?
Thanks

**Comment by frankist** [comment, marked as answer]
Hey, we concluded recently that there can be more than one PDCCH for the same UE per slot. This is important to support more UL slots than DL slots. However, the code is not yet reflecting those findings. We will change this soon.

**Reply by frankist** [reply]
However, the claim:
> Going through scheduler_time_rr.cpp we have found these two code sections that hints at the fact that one cannot schedule PDCCH and PDSCH in the same slot.

is not correct. We support the same UE being scheduled in PDCCH and PDSCH in the same slot.
